syslog/syslog.go

```diff
@@ -60,9 +60,12 @@ func dial(network, raddr string, rootCAs *x509.CertPool) (*conn, error) {
 		if rootCAs != nil {
 			config = &tls.Config{RootCAs: rootCAs}
 		}
-		netConn, err = tls.Dial("tcp", raddr, config)
+		dialer := &net.Dialer{
+			Timeout : time.Duration(30) * time.Second,
+		}
+		netConn, err = tls.DialWithDialer(dialer, "tcp", raddr, config)
 	case "udp", "tcp":
-		netConn, err = net.Dial(network, raddr)
+		netConn, err = net.DialTimeout(network, raddr, time.Duration(30) * time.Second)
```

Comment:
I didn't know this when I wrote the code, but it sounds like we should keep this in even for UDP as it [also covers name resolution](https://github.com/golang/go/blob/f78a4c84ac8ed44aaf331989aa32e40081fd8f13/src/net/dial.go#L190).